src/main/java/me/twostinkysocks/boxplugin/manager/MarketManager.java

```diff
@@ -22,11 +22,15 @@
 
 public class MarketManager {
 
+    private HashMap<UUID, Integer> selectedCoins = new HashMap<>();
+
     public void openGui(Player p) {
         ChestGui gui = new ChestGui(3, "Market");
         StaticPane pane = new StaticPane(9,3);
         gui.setOnGlobalClick(e -> e.setCancelled(true));
 
+        selectedCoins.put(p.getUniqueId(), 0);
```

Comment:
why not put this in openWithDrawMenu, since if they open deposit it will not be cleared